src/Reader/Reader.php

```diff
@@ -117,23 +118,30 @@ public static function setCache(CacheStorage $cache)
      *
      * Sets the HTTP client object to use for retrieving the feeds.
      *
-     * @param  ZendHttp\Client $httpClient
+     * @param  ZendHttp\Client | Http\ClientInterface $httpClient
      * @return void
      */
-    public static function setHttpClient(ZendHttp\Client $httpClient)
+    public static function setHttpClient($httpClient)
     {
+        if ($httpClient instanceof ZendHttp\Client) {
+            $httpClient = new Http\ZendHttpClientDecorator($httpClient);
+        }
+
+        if (! $httpClient instanceof Http\ClientInterface) {
+            throw new InvalidHttpClientException();
+        }
         static::$httpClient = $httpClient;
     }
 
     /**
      * Gets the HTTP client object. If none is set, a new ZendHttp\Client will be used.
      *
-     * @return ZendHttp\Client
+     * @return Http\ClientInterface
```

Comment:
We can't; method conflicts.

While it's a potential BC break, we've been messaging since 2.3 that `(set|get)HttpClient()` are *supposed* to allow injecting alternate clients. This typehint was essentially missed when we made that change. Additionally, this is a method that is largely never used; you inject the HTTP client, and then consume the `Reader` instance.

To help mitigate the change, however, I've added a `getDecoratedClient()` method to the `ZendHttpClientDecorator`. This should be sufficient for messaging to users for the upcoming minor release.